commands/apps.go

```diff
@@ -393,6 +403,69 @@ func RunAppsDelete(c *CmdConfig) error {
 	return nil
 }
 
+// RunAppsDetect detects an function project and converts it into apps project.
+func RunAppsDetect(c *CmdConfig) error {
+	source, err := c.Doit.GetString(c.NS, doctl.ArgProjectSource)
+	if err != nil {
+		return err
+	}
+	if len(source) == 0 {
+		return fmt.Errorf("source cannot be empty")
+	}
+
+	sha, err := c.Doit.GetString(c.NS, doctl.ArgCommitHash)
+	if err != nil {
+		return err
+	}
+
+	name, err := c.Doit.GetString(c.NS, doctl.ArgProjectName)
+	if err != nil {
+		return err
+	}
+	if len(name) == 0 {
+		return fmt.Errorf("name cannot be empty")
+	}
+
+	branch, err := c.Doit.GetString(c.NS, doctl.ArgProjectBrach)
+	if err != nil {
+		return err
+	}
+	if len(branch) == 0 {
+		return fmt.Errorf("branch cannot be empty")
+	}
+
+	// Need to check, How user value will be overrided
+	autoDeploy, err := c.Doit.GetBool(c.NS, doctl.ArgDeployOnPush)
+	if err != nil {
+		return err
+	}
+	if len(c.Args) > 0 {
```

Comment:
I don't think we need argument parsing here if we are including `auto-deploy` as a flag as well.